commands/util.go

```diff
@@ -135,44 +140,47 @@ func GetGroupID(owner string) int {
 }
 
 // CompressData compresses and base64 encodes a string to place into Consul's KV store.
-func CompressData(data string) string {
+func CompressData(data string) []byte {
 	var compressed bytes.Buffer
 	gz, _ := gzip.NewWriterLevel(&compressed, gzip.BestCompression)
 	gz.Write([]byte(data))
 	gz.Flush()
 	gz.Close()
-	encoded := base64.StdEncoding.EncodeToString(compressed.Bytes())
+	encoded := compressed.Bytes()
 	Log(fmt.Sprintf("compressing='true' full_size='%d' compressed_size='%d'", len(data), len(encoded)), "info")
 	return encoded
 }
 
 // DecompressData base64 decodes and decompresses a string taken from Consul's KV store.
-func DecompressData(data string) string {
-	if data != "" {
-		// If it's been compressed, it's been base64 encoded.
-		raw, err := base64.StdEncoding.DecodeString(data)
+func DecompressData(data []byte) string {
+	if data == nil {
+		return ""
+	}
+	// If it's been compressed, it's may have been base64 encoded (previous 1.16).
+	if rxBase64.Match(data) {
```

Comment:
Why not rather use the flags from consul? Is it already used? 
It would allow to support further updates if needed (like compression algorithm updates)

Reply:
+1 for flags :)
See: https://github.com/DataDog/kvexpress/pull/110/commits/5fc283e18841a23242d725d2102a141d12882d0f